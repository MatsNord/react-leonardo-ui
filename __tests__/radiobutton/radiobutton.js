import React from 'react';
import renderer from 'react-test-renderer';

import Radiobutton from '../../src/radio-button';

test('Normal radiobutton renders correctly', () => {
  const tree = renderer.create(
    <Radiobutton checked />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Radiobutton with states renders correctly', () => {
  const tree = renderer.create(
    <Radiobutton checked active disabled />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Radiobutton with modifiers renders correctly', () => {
  const tree = renderer.create(
    <Radiobutton checked variant="inverse" />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Radiobutton passes props correctly', () => {
  const onChange = jest.fn(() => {});

  const tree = renderer.create(
    <Radiobutton checked className="my-class" data-x="x" title="Title" onChange={onChange} />
  ).toJSON();

  const input = tree.children[0];
  input.props.onChange();
  expect(onChange.mock.calls.length).toBe(1);
});
