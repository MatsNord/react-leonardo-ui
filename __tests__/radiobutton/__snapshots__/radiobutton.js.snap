exports[`test Normal radiobutton renders correctly 1`] = `
<label
  className="lui-radiobutton"
  htmlFor={undefined}>
  <input
    checked={true}
    className="lui-radiobutton__input"
    type="radio" />
  <div
    className="lui-radiobutton__radio-wrap">
    <span
      className="lui-radiobutton__radio" />
    <span
      className="lui-radiobutton__radio-text" />
  </div>
</label>
`;

exports[`test Radiobutton with modifiers renders correctly 1`] = `
<label
  className="lui-radiobutton lui-radiobutton--inverse"
  htmlFor={undefined}>
  <input
    checked={true}
    className="lui-radiobutton__input"
    type="radio" />
  <div
    className="lui-radiobutton__radio-wrap">
    <span
      className="lui-radiobutton__radio" />
    <span
      className="lui-radiobutton__radio-text" />
  </div>
</label>
`;

exports[`test Radiobutton with states renders correctly 1`] = `
<label
  className="lui-radiobutton lui-active"
  htmlFor={undefined}>
  <input
    checked={true}
    className="lui-radiobutton__input"
    disabled={true}
    type="radio" />
  <div
    className="lui-radiobutton__radio-wrap">
    <span
      className="lui-radiobutton__radio" />
    <span
      className="lui-radiobutton__radio-text" />
  </div>
</label>
`;
